### ISSUE TITLE
x509: certificate signed by unknown authority (possibly because of "x509: invalid signature: parent certificate cannot sign this kind of certificate" while trying to verify candidate authority certificate

### ISSUE THREAD

**Issue by cesarsaez** [cesarsaez/ansible-role-opensslcaandcerts#1]
openssl s_client -showcerts -connect vault01:8200
```
CONNECTED(00000003)
Can't use SSL_get_servername
depth=1 C = ES, O = ACME SOLUTIONS, CN = VAULT-CA
verify error:num=19:self signed certificate in certificate chain
verify return:1
depth=1 C = ES, O = ACME SOLUTIONS, CN = VAULT-CA
verify error:num=24:invalid CA certificate
verify return:1
depth=1 C = ES, O = ACME SOLUTIONS, CN = VAULT-CA
verify error:num=26:unsupported certificate purpose
verify return:1
depth=1 C = ES, O = ACME SOLUTIONS, CN = VAULT-CA
verify return:1
depth=0 C = ES, O = ACME Solutions, OU = IT LABS, CN = vaultclus.gaseosalab.net
verify return:1
---
Certificate chain
 0 s:C = ES, O = ACME Solutions, OU = IT LABS, CN = vaultclus.gaseosalab.net
   i:C = ES, O = ACME SOLUTIONS, CN = VAULT-CA
```

**Comment by cesarsaez** [comment]
they CA certificate has no setting regarding the Key Usage, that can be related to "unsupported certificate purpose"

**Comment by cesarsaez** [comment]
Previous debug shown 2 actual issues:
- error:num=26:unsupported certificate purpose // meaning missing proper key_usage
- error:num=19:self signed certificate in certificate chain // meaning that its a self-signed certinitica becuase its missing the **basic_constraints: "CA:TRUE"** (the TRUE must be in capital as per https://fossies.org/linux/ansible/lib/ansible/module_utils/crypto.py look for "CA:")

Fixed at 1dfec10f70806932b1b37ea115f748bbcd728d0b
